packages/components/src/Gesture/withListGesture.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import keycode from 'keycode';
+import omit from 'lodash/omit';
+
+function focusOn(event, element) {
+	if (element) {
+		element.focus();
+	}
+}
+
+function getAllItems(ref) {
+	return ref.closest('[role="list"]').querySelectorAll('[role="listitem"]');
+}
+
+function getNextItem(ref) {
+	let nextElement;
+	let currentFound;
+	let hasNext;
+
+	const nodes = getAllItems(ref).values();
+
+	do {
+		const { value, done } = nodes.next();
+
+		if (currentFound) {
+			nextElement = value;
+			hasNext = false;
+		} else {
+			currentFound = value === ref;
+			hasNext = !done;
+		}
+	} while (hasNext);
+
+	return nextElement;
+}
+
+function getPreviousItem(ref) {
+	let previousElement;
+	let hasNext;
+
+	const nodes = getAllItems(ref).values();
+
+	do {
+		const { value, done } = nodes.next();
+		const currentFound = value === ref;
+
+		if (currentFound) {
+			hasNext = false;
+		} else {
+			previousElement = value;
+			hasNext = !done;
+		}
+	} while (hasNext);
+
+	return previousElement;
+}
+
+function onKeyDown(event, ref) {
+	switch (event.keyCode) {
+		case keycode.codes.down:
+			event.stopPropagation();
+			focusOn(event, getNextItem(ref));
+			break;
+		case keycode.codes.up:
+			event.stopPropagation();
+			focusOn(event, getPreviousItem(ref));
+			break;
+		default:
+			break;
+	}
+}
+
+export default function withListGesture(WrappedComponent) {
+	function ListGesture(props) {
+		return <WrappedComponent {...props} lol={'lol'} onKeyDown={onKeyDown} />;
+	}
+
+	ListGesture.propTypes = {
+		...omit(WrappedComponent.propTypes, 'onKeyDown'),
```

Comment:
Is it necessary to defined the `propTypes` of the `wrappedComponent` if they are not used in this layer ?

Reply:
Good point, I remove it.